modules/integration/tests-integration/tests-amqp/src/test/java/org/wso2/mb/integration/tests/amqp/functional/TenantDeadLetterChannelTestCase.java

```diff
@@ -0,0 +1,214 @@
+            System.clearProperty("AndesAckWaitTimeOut");
+        } else {
+            System.setProperty("AndesAckWaitTimeOut", defaultAndesAckWaitTimeOut);
+        }
+    }
+
+
+    /**
+     * This test case will test functionality of tenant dead letter channel.
+     * 1. Publish 1 queue message to tenant.
+     * 2. Add consumer for the queue message.
+     * 3. Consumer do not acknowledge for the queue message.
+     * 4. Message will put into tenant dlc after retry sending queue message 10 times.
+     * 5. Number of messages in tenant dlc should be equal to 1.
+     * 6. Number of messages in super tenant dlc should be equal to 0.
+     *
+     * @throws JMSException
+     * @throws IOException
+     * @throws NamingException
+     * @throws AndesClientConfigurationException
+     * @throws AndesClientException
+     * @throws LoginAuthenticationExceptionException
+     * @throws XPathExpressionException
+     * @throws AndesAdminServiceBrokerManagerAdminException
+     * @throws URISyntaxException
+     * @throws SAXException
+     * @throws LogoutAuthenticationExceptionException
+     * @throws XMLStreamException
+     */
+    @Test(groups = "wso2.mb", description = "Tenant dead letter channel test case")
+    public void performTenantDeadLetterChannelTestCase()
+            throws JMSException, IOException, NamingException, AndesClientConfigurationException,
+                   AndesClientException, LoginAuthenticationExceptionException,
+                   XPathExpressionException,
+                   AndesAdminServiceBrokerManagerAdminException, URISyntaxException, SAXException,
+                   LogoutAuthenticationExceptionException, XMLStreamException {
+
+        int sendMessageCount = 1;
+
+        String destinationName = "dlctenant1.com/tenantQueue";
+
+        // Create a consumer client configuration
+        AndesJMSConsumerClientConfiguration consumerConfig =
+                new AndesJMSConsumerClientConfiguration("dlctenantuser1!dlctenant1.com",
+                                                        "dlctenantuser1", ExchangeType.QUEUE,
+                                                        destinationName);
+        // Add manual client acknowledgement in configuration
+        consumerConfig
+                .setAcknowledgeMode(JMSAcknowledgeMode.CLIENT_ACKNOWLEDGE);
+        // Acknowledge a message only after 200 messages are received
+        consumerConfig
+                .setAcknowledgeAfterEachMessageCount(200L);
+        consumerConfig.setPrintsPerMessageCount(sendMessageCount);
+
+        // Create consumer client with given consumerConfig
+        AndesClient consumerClient = new AndesClient(consumerConfig, true);
+        // Start consumer client
+        consumerClient.startClient();
+
+        // Create a publisher client configuration
+        AndesJMSPublisherClientConfiguration tenantPublisherConfig =
+                new AndesJMSPublisherClientConfiguration("dlctenantuser1!dlctenant1.com",
```

Comment:
currently there are no methods available in test suite to refer tenant domains/users from automation.xml to test cases. we can do this as improvement to whole integration test suite.